### ISSUE TITLE
Components setup

### ISSUE THREAD

**Issue by touficbatache** [material-components/material-components-web#3999]
Hi,
I've been trying for the past hour to migrate [my website](https://www.toufic.ml/index.php) from an old version of MDC to the latest one. But in the process, I've noticed that so many things have changed and I couldn't setup the Drawer nor the Tab Bar component. The new documentation is just so confusing. I'm using pure JS, but nothing's working. The old demos were much better as we could see how it's done, whereas here it's a big struggle. Can anyone provide a codepen or something demonstrating how to use the modal drawer and the tab bar component?
Thanks

**Comment by acdvorak** [comment]
Thanks for the constructive feedback Toufic, it's much appreciated!

### TL;DR

You can find plain JS examples split across two directories:

https://github.com/material-components/material-components-web/tree/master/demos
https://github.com/material-components/material-components-web/tree/master/test/screenshot/spec

### Why?

The old demo pages attempted to satisfy three different needs:

1. Let users interact with our components and see how the different variants look/behave
2. Demonstrate how to use the plain vanilla JS APIs
3. Allow MCD Web team members to manually test our work

It became clear that one page could not do all those things (at least not very well), so we decided to split the demo pages up into a simplified [public Catalog site](https://material-components.github.io/material-components-web-catalog/#/) and more thorough [screenshot test pages](https://storage.googleapis.com/mdc-web-screenshot-tests/travis/2018/10/26/20_56_32_028/report/report.html?utm_source=github&utm_medium=issue).

However, we're currently in the middle of that transition, so not all components or variants have screenshot test pages.

IIRC some variants of the new [`MDCTopAppBar`](https://github.com/material-components/material-components-web/tree/v0.40.1/packages/mdc-top-app-bar) and [`MDCTabBar`](https://github.com/material-components/material-components-web/tree/v0.40.1/packages/mdc-tab-bar) packages can technically be used together, but we don't _officially_ support that combination yet, so it might take a little fiddling on your end to make it look right until we have a chance to implement #3633.

### Documentation

I agree with you that our documentation needs to be improved. We're actively investigating this, and looking at ideas like providing inline examples and live playgrounds directly within the docs.
Unfortunately, that's a longer-term effort.

**Comment by touficbatache** [comment]
I would suggest adding that the `<div class="mdc-drawer-scrim"></div>` element is necessary for the modal drawer to work. This is what was breaking everything! Thanks

**Comment by kfranqueiro** [comment]
We have a note under the modal example, but I've sent a PR to update its wording.

RE the state of demos / catalog / screenshot tests, I agree that we're definitely not in an ideal state. There's some discussion/planning being done in terms of efforts in this area going into 2019.
